tests/test_client.py

```diff
@@ -312,3 +341,74 @@ def ListDefinitions(self, _, context):
     received_metadata_dict = dict(received_metadata[0])
     assert received_metadata_dict["token"] == "007"
     assert received_metadata_dict["identity"] == "james bond"
+
+
+def test_not_configured():
+    with pytest.raises(pypim.NotConfiguredError):
+        pypim.connect()
+
+
+@pytest.mark.parametrize(
+    ("bad_configuration,message_content"),
+    [
+        (r"""not even the right format""", "json"),
+        (r"""{"version": 2, "pim": "future format"}""", "Unsupported version"),
+        (
+            r"""{"version": 1, "pim": {
+                "headers": {"token": "007","identity": "james bond"},"tls": false}}""",
+            "uri",
+        ),
+        (r"""{"version": 1, "pim": {"uri": "dns:127.0.0.1:5000","tls": false}}""", "headers"),
+        (
+            r"""{"version": 1, "pim": {"uri": "dns:127.0.0.1:5000",
+            "headers": {"token": "007","identity": "james bond"}}}""",
+            "tls",
+        ),
+        (
+            r"""{"version": 1, "pim": {"uri": "dns:127.0.0.1:5000", "tls": true,
+            "headers": {"token": "007","identity": "james bond"}}}""",
+            "not yet supported",
+        ),
+    ],
+)
+def test_bad_configuration(tmp_path, bad_configuration, message_content):
+    config_path = tmp_path / "pim.json"
+    with open(config_path, "w") as f:
+        f.write(bad_configuration)
+
+    with pytest.raises(pypim.InvalidConfigurationError) as exc:
+        pypim.Client._from_configuration(config_path)
+
+    assert message_content in str(exc)
+
+
+def test_list_instance_error(
+    testing_pool: ThreadPoolExecutor,
+    testing_channel: grpc_testing.Channel,
+):
+    # Arrange
+    # A server serving a 500 on ListInstance and ListDefinitions
+    def server():
+        _, update_request, rpc = testing_channel.take_unary_unary(LIST_INSTANCES_METHOD)
+        rpc.terminate(None, [], StatusCode.INTERNAL, "I'm a teapot")
+        _, update_request, rpc = testing_channel.take_unary_unary(LIST_DEFINITIONS_METHOD)
+        rpc.terminate(None, [], StatusCode.INTERNAL, "I'm a teapot")
+        return update_request
+
+    testing_pool.submit(server)
+    client = pypim.Client(testing_channel)
+
+    # Act
+    # List the instances and the definitions
+    with pytest.raises(pypim.RemoteError) as exc1:
+        client.instances()
+    with pytest.raises(pypim.RemoteError) as exc2:
+        client.definitions()
+
+    # Assert
+    # The user got the server message
+    assert "I'm a teapot" in str(exc1)
+    assert "I'm a teapot" in str(exc2)
```

Comment:
I think periods should go in the quote "I'm a teapot." However, I wasn't sure, so I didn't add them here.

Reply:
```suggestion
    assert "I'm a teapot." in str(exc1)
    assert "I'm a teapot." in str(exc2)
```

These lines test that the error is correctly transferred, so it makes sense to add the period here too